File: 7_binary_search/이준우/입국심사.py
def solution(n, times):
    answer = min(times) * n
    
    start = 1
    end = answer
    while start <= end:
        mid = (start + end) // 2
        cnt = sum(mid // time for time in times)
        
        if cnt < n:
            start = mid + 1
        else:
            answer = min(answer, mid)
            end = mid - 1

    return answer

File: 7_binary_search/김민주/입국심사.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
using namespace std;

long long solution(int n, vector<int> times) {
    long long answer = 0;
    long long m = times.size();
    
    sort(times.begin(), times.end());
    long long start =1;
    long long end = (long long)n*times[m-1];
    long long mid = (start+end)/2;
    long long visitor = 0;
    
    while(start <= end){
        visitor = 0;
        mid = (start+end)/2;
        
        for (int i=0; i<m; i++){
            visitor += (mid / times[i]);
        }
        
        if (visitor >= n){
            answer = mid;
            end = mid-1;
            // break;
        }else{
            start = mid+1;
        }
    }
    
    
    return answer;
}